browser-test/src/admin/admin_program_migration.test.ts

```diff
@@ -266,12 +258,63 @@ test.describe('program migration', () => {
       await expect(programDataDiv).toContainText('Garlic Press')
     })
 
-    await test.step('save the imported program', async () => {
-      await adminProgramMigration.saveProgram()
-      await adminPrograms.expectProgramExist(
-        'Comprehensive Sample Program 2',
-        'desc',
+    await test.step('delete the program and start over without saving', async () => {
+      await adminProgramMigration.clickButton('Delete and start over')
+      await adminProgramMigration.expectImportPage()
+      await expect(page.getByRole('textbox')).toHaveValue('')
+    })
+
+    await test.step('attempt to save the program and see error', async () => {
+      await adminProgramMigration.submitProgramJson(
+        downloadedComprehensiveProgram,
+      )
+      await adminProgramMigration.clickButton('Save')
+      // a db error is thrown because there is already a draft of this program
+      await adminProgramMigration.expectAlert(
+        'Unable to save program. Please try again or contact your IT team for support.',
+      )
+      await validateScreenshot(page, 'error-saving-program')
+    })
+
+    await test.step('publish the program and attempt to import it again', async () => {
+      await adminPrograms.gotoAdminProgramsPage()
+      await adminPrograms.publishAllDrafts()
+      await adminPrograms.gotoExportProgramPage(
+        'Comprehensive Sample Program',
+        'ACTIVE',
+      )
+      downloadedComprehensiveProgram =
+        await adminProgramMigration.downloadJson()
+      await adminPrograms.gotoAdminProgramsPage()
+      await adminProgramMigration.goToImportPage()
+      await adminProgramMigration.submitProgramJson(
+        downloadedComprehensiveProgram,
+      )
+      await adminProgramMigration.clickButton('Save')
+      // the save is successful because no draft already exists
+      await adminProgramMigration.expectAlert(
+        'Your program has been successfully imported',
+      )
+      await validateScreenshot(page, 'saved-program-success')
+    })
+
+    await test.step('import another program and go directly to program edit page', async () => {
+      await adminProgramMigration.clickButton('Import another program')
+      await adminProgramMigration.expectImportPage()
+      await expect(page.getByRole('textbox')).toHaveValue('')
+      // replace the question admin name to avoid draft collision
```

Comment:
**suggestion (non-blocking):** You could consider making this and the comment below a test.step too

Reply:
done!